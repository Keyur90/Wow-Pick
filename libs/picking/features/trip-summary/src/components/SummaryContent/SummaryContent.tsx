import React from 'react';
import { startCase as _startCase } from 'lodash';
import { makeStyles } from '@material-ui/core/styles';
import { List, ListItem, BagStatus } from '@rf2/ui';
import { textDuration } from '@rf2/shared/utility';
import { LabelIcon, ToteIcon, TimerIcon } from '@rf2/shared/icons';
import { LabelledValue } from '../LabelledValue';

interface SummaryContentPropTypes {
  isExpressTrip: boolean;
  isBulkTrip: boolean;
  totes: number;
  articles: number;
  labels: number;
  goalTime: number;
  bagType?: string;
}

const useStyles = makeStyles((theme) => ({
  containerWrapper: {
    backgroundColor: theme.palette.common.white,
  },
  bagType: {
    fontSize: '1rem',
  },
}));

const SummaryContent: React.FC<SummaryContentPropTypes> = ({
  isExpressTrip,
  isBulkTrip,
  totes,
  articles,
  labels,
  goalTime,
  bagType,
}) => {
  const classes = useStyles();

  return (
    <>
      <div className={classes.containerWrapper}>
        <List>
          {isBulkTrip && <ListItem>Bulk Trolley Trip</ListItem>}
          {!isExpressTrip && (
            <>
              <ListItem>
                <ToteIcon />
                <LabelledValue label="Totes" value={totes} />
                <LabelledValue label="Articles" value={articles} />
              </ListItem>
              <ListItem>
                <LabelIcon />
                <LabelledValue label="Labels" value={labels} />
              </ListItem>
            </>
          )}
          {isExpressTrip && (
            <>
              <ListItem>
                <LabelledValue label="Articles" value={articles} />
              </ListItem>
              <ListItem>No tote labels needed</ListItem>
              <ListItem>
                <LabelledValue
                  label="Bag type"
                  value={
                    <span className={classes.bagType}>
                      <BagStatus bagType={bagType} />
                    </span>
                  }
                />
              </ListItem>
            </>
          )}
        </List>
      </div>
      <div className={classes.containerWrapper}>
        <List>
          <ListItem>
            <TimerIcon />
            <LabelledValue label="Goal Time" value={textDuration(goalTime)} />
          </ListItem>
        </List>
      </div>
    </>
  );
};

export { SummaryContent };
