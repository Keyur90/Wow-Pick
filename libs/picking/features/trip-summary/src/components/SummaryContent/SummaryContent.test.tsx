import React from 'react';
import { render } from '@testing-library/react';
import { SummaryContent } from './SummaryContent';

test('renders SummaryContent', () => {
  const { container } = render(
    <SummaryContent isBulkTrip={false} totes={1} articles={24} labels={1} goalTime={360} isExpressTrip={false} />
  );

  expect(container).toMatchSnapshot();
});

test('renders SummaryContent bulk trip', () => {
  const { getByText } = render(
    <SummaryContent isBulkTrip={true} totes={1} articles={24} labels={1} goalTime={360} isExpressTrip={false} />
  );

  expect(getByText(/Bulk trolley trip/i)).toBeInTheDocument();
});
