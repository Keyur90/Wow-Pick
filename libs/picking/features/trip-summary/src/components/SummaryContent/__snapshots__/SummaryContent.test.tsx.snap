// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SummaryContent 1`] = `
<div>
  <div
    class="makeStyles-containerWrapper-1"
  >
    <div
      class="makeStyles-root-3"
    >
      <div
        class="MuiContainer-root makeStyles-root-5 MuiContainer-maxWidthLg"
      >
        <ul
          class="makeStyles-list-4"
        >
          <li>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21.7664 7.50187H20.5522L14.7733 1.71938C14.481 1.42687 14.0087 1.42687 13.7164 1.71938C13.4241 2.01188 13.4241 2.48438 13.7164 2.77688L18.431 7.49436H5.56153L10.2761 2.77688C10.5684 2.48438 10.5684 2.01188 10.2761 1.71938C9.98377 1.42687 9.51156 1.42687 9.21925 1.71938L3.44035 7.50187H2.22611C0.996877 7.50187 0 8.49937 0 9.77437C0 10.7944 0.68957 11.6569 1.64147 11.9194L3.77014 20.7244C3.9875 21.7594 4.91694 22.5019 5.96628 22.5019H18.0262C19.0831 22.5019 20.005 21.7519 20.2224 20.7394L22.351 11.9194C23.2954 11.6569 23.9925 10.7869 23.9925 9.72186C23.9925 8.49936 22.9956 7.50187 21.7664 7.50187ZM21.7664 10.5019H21.7439C21.3991 10.5019 21.0993 10.7419 21.0169 11.0794L18.7608 20.4094C18.6858 20.7544 18.3785 21.0019 18.0262 21.0019H5.96628C5.614 21.0019 5.30669 20.7544 5.23174 20.3869L2.98314 11.0719C2.90069 10.7344 2.5784 10.4944 2.23361 10.4944C1.83636 10.4944 1.50656 10.1719 1.50656 9.72186C1.50656 9.32436 1.82886 8.99436 2.23361 8.99436H21.7664C22.1636 8.99436 22.4934 9.31686 22.4934 9.76686C22.4934 10.1794 22.1711 10.5019 21.7664 10.5019Z"
                fill="#3A474E"
              />
              <path
                d="M11.9813 12.0098C11.7713 12.0098 11.6064 12.1747 11.6064 12.3845V18.3789C11.6064 18.5812 11.7713 18.7535 11.9813 18.7535C12.1837 18.7535 12.3561 18.5812 12.3561 18.3789V12.3845C12.3561 12.1747 12.1837 12.0098 11.9813 12.0098Z"
                fill="#3A474E"
              />
              <path
                d="M14.9353 12.0473C14.7329 12.0248 14.5454 12.1747 14.5229 12.377L13.8931 18.3414C13.8707 18.5437 14.0206 18.731 14.2305 18.7535C14.433 18.776 14.6204 18.6261 14.6429 18.4163L15.2727 12.4519C15.2877 12.2571 15.1378 12.0697 14.9353 12.0473Z"
                fill="#3A474E"
              />
              <path
                d="M17.6495 12.1147C17.447 12.0773 17.2521 12.2121 17.2146 12.4219L16.1724 18.3264C16.135 18.5287 16.2699 18.7235 16.4799 18.761C16.6823 18.7985 16.8772 18.6636 16.9147 18.4538L17.9569 12.5493C17.9869 12.347 17.8519 12.1447 17.6495 12.1147Z"
                fill="#3A474E"
              />
              <path
                d="M9.02717 12.0473C8.82474 12.0697 8.67478 12.2571 8.69728 12.4594L9.32708 18.4238C9.34957 18.6261 9.537 18.776 9.73944 18.761C9.94187 18.7385 10.0918 18.5512 10.0693 18.3489L9.43954 12.3845C9.41704 12.1747 9.22961 12.0248 9.02717 12.0473Z"
                fill="#3A474E"
              />
              <path
                d="M6.30553 12.1147C6.10309 12.1522 5.96813 12.347 5.99812 12.5493L7.04031 18.4538C7.0778 18.6561 7.27273 18.791 7.47516 18.761C7.6776 18.7235 7.81256 18.5287 7.78257 18.3264L7.26148 15.3742L6.74038 12.4219C6.7029 12.2121 6.50797 12.0773 6.30553 12.1147Z"
                fill="#3A474E"
              />
            </svg>
            <span>
              <span
                class="makeStyles-label-6"
              >
                Totes
                :
              </span>
              <span
                class="makeStyles-value-7"
              >
                1
              </span>
            </span>
            <span>
              <span
                class="makeStyles-label-6"
              >
                Articles
                :
              </span>
              <span
                class="makeStyles-value-7"
              >
                24
              </span>
            </span>
          </li>
          <li>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 18"
            >
              <path
                d="M21.75 0H2.25C1.0125 0 0 1.0125 0 2.25V15.75C0 16.9875 1.0125 18 2.25 18H21.75C22.9875 18 24 16.9875 24 15.75V2.25C24 1.005 22.9875 0 21.75 0ZM22.5 15.75C22.5 16.1625 22.1625 16.5 21.75 16.5H2.25C1.8375 16.5 1.5 16.1625 1.5 15.75V7.5H22.5V15.75ZM22.5 7.125H1.5V2.25C1.5 1.8375 1.8375 1.5 2.25 1.5H21.75C22.1625 1.5 22.5 1.8375 22.5 2.25V7.125Z"
                fill="#3A474E"
              />
              <path
                d="M3 3.75H5.25C5.46 3.75 5.625 3.585 5.625 3.375C5.625 3.165 5.46 3 5.25 3H3C2.7975 3 2.625 3.165 2.625 3.375C2.625 3.585 2.7975 3.75 3 3.75Z"
                fill="#3A474E"
              />
              <path
                d="M3 6H8.25C8.46 6 8.625 5.835 8.625 5.625C8.625 5.415 8.46 5.25 8.25 5.25H5.625H3C2.7975 5.25 2.625 5.415 2.625 5.625C2.625 5.835 2.7975 6 3 6Z"
                fill="#3A474E"
              />
              <path
                d="M8.5875 9C8.44575 9 8.325 9.11 8.325 9.25V14.75C8.325 14.885 8.44575 15 8.5875 15C8.72925 15 8.85 14.885 8.85 14.75V9.25C8.85 9.11 8.72925 9 8.5875 9Z"
                fill="#3A474E"
              />
              <path
                d="M11.7375 9C11.5957 9 11.475 9.11 11.475 9.25V14.75C11.475 14.885 11.5957 15 11.7375 15C11.8792 15 12 14.885 12 14.75V9.25C12 9.11 11.8792 9 11.7375 9Z"
                fill="#3A474E"
              />
              <path
                d="M12.7875 9C12.6457 9 12.525 9.11 12.525 9.25V14.75C12.525 14.885 12.6457 15 12.7875 15C12.9292 15 13.05 14.885 13.05 14.75V9.25C13.05 9.11 12.9292 9 12.7875 9Z"
                fill="#3A474E"
              />
              <path
                d="M15.4125 9C15.2708 9 15.15 9.11 15.15 9.25V14.75C15.15 14.885 15.2708 15 15.4125 15C15.5542 15 15.675 14.885 15.675 14.75V9.25C15.675 9.11 15.5542 9 15.4125 9Z"
                fill="#3A474E"
              />
              <path
                d="M9.9 9C9.61125 9 9.375 9.225 9.375 9.5V14.5C9.375 14.775 9.61125 15 9.9 15C10.1887 15 10.425 14.775 10.425 14.5V9.5C10.425 9.225 10.1887 9 9.9 9Z"
                fill="#3A474E"
              />
              <path
                d="M7.275 9C6.98625 9 6.75 9.225 6.75 9.5V14.5C6.75 14.775 6.98625 15 7.275 15C7.56375 15 7.8 14.775 7.8 14.5V9.5C7.8 9.225 7.56375 9 7.275 9Z"
                fill="#3A474E"
              />
              <path
                d="M14.1 9C13.8112 9 13.575 9.225 13.575 9.5V14.5C13.575 14.775 13.8112 15 14.1 15C14.3887 15 14.625 14.775 14.625 14.5V9.5C14.625 9.225 14.3887 9 14.1 9Z"
                fill="#3A474E"
              />
              <path
                d="M16.725 15C17.0137 15 17.25 14.775 17.25 14.5V12V9.5C17.25 9.225 17.0137 9 16.725 9C16.4362 9 16.2 9.225 16.2 9.5V14.5C16.2 14.775 16.4362 15 16.725 15Z"
                fill="#3A474E"
              />
            </svg>
            <span>
              <span
                class="makeStyles-label-6"
              >
                Labels
                :
              </span>
              <span
                class="makeStyles-value-7"
              >
                1
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-containerWrapper-1"
  >
    <div
      class="makeStyles-root-3"
    >
      <div
        class="MuiContainer-root makeStyles-root-5 MuiContainer-maxWidthLg"
      >
        <ul
          class="makeStyles-list-4"
        >
          <li>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 25 24"
            >
              <path
                d="M15.99 13.5375L12 12.21V5.25C12 4.8375 11.6625 4.5 11.25 4.5C10.8375 4.5 10.5 4.8375 10.5 5.25V13.29L15.51 14.9625C15.585 14.9925 15.6675 15 15.75 15C16.065 15 16.3575 14.7975 16.4625 14.49C16.59 14.0925 16.38 13.6725 15.99 13.5375Z"
                fill="#3A474E"
              />
              <path
                d="M23.7825 12.9675C23.49 12.675 23.0175 12.675 22.725 12.9675L22.5 13.1925V12C22.5 5.7975 17.4525 0.75 11.25 0.75C5.0475 0.75 0 5.7975 0 12C0 17.25 3.72001 21.87 8.84251 22.9875C9.24751 23.0775 9.64501 22.8225 9.73501 22.4175C9.83251 22.0125 9.57 21.615 9.165 21.525C4.7175 20.5575 1.5 16.5525 1.5 12C1.5 6.6225 5.8725 2.25 11.25 2.25C16.6275 2.25 21 6.6225 21 12V13.1925L20.7825 12.975C20.49 12.6825 20.0175 12.6825 19.725 12.975C19.575 13.125 19.5075 13.3125 19.5075 13.5075C19.5075 13.7025 19.5825 13.89 19.725 14.04L21.225 15.54C21.5175 15.8325 21.99 15.8325 22.2825 15.54L23.7825 14.04C24.075 13.74 24.075 13.26 23.7825 12.9675Z"
                fill="#3A474E"
              />
              <path
                d="M20.25 18.75C20.6642 18.75 21 18.4142 21 18C21 17.5858 20.6642 17.25 20.25 17.25C19.8358 17.25 19.5 17.5858 19.5 18C19.5 18.4142 19.8358 18.75 20.25 18.75Z"
                fill="#3A474E"
              />
              <path
                d="M18 21C18.4142 21 18.75 20.6642 18.75 20.25C18.75 19.8358 18.4142 19.5 18 19.5C17.5858 19.5 17.25 19.8358 17.25 20.25C17.25 20.6642 17.5858 21 18 21Z"
                fill="#3A474E"
              />
              <path
                d="M15 22.5C15.4142 22.5 15.75 22.1642 15.75 21.75C15.75 21.3358 15.4142 21 15 21C14.5858 21 14.25 21.3358 14.25 21.75C14.25 22.1642 14.5858 22.5 15 22.5Z"
                fill="#3A474E"
              />
              <path
                d="M12 23.25C12.4142 23.25 12.75 22.9142 12.75 22.5C12.75 22.0858 12.4142 21.75 12 21.75C11.5858 21.75 11.25 22.0858 11.25 22.5C11.25 22.9142 11.5858 23.25 12 23.25Z"
                fill="#3A474E"
              />
            </svg>
            <span>
              <span
                class="makeStyles-label-6"
              >
                Goal Time
                :
              </span>
              <span
                class="makeStyles-value-7"
              >
                6 minutes
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
